test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs

```diff
@@ -38,23 +74,131 @@ public void DthStartup_GetProjectInformation(string flavor, string os, string ar
             {
                 client.Initialize(testProject);
 
-                var response = client.GetResponse<ProjectMessage>();
-                Assert.NotNull(response);
-                Assert.Equal(server.HostId, response.HostId);
-                Assert.Equal(0, response.ContextId);
-                Assert.Equal("ProjectInformation", response.MessageType);
+                var response = client.DrainTillFirst("ProjectInformation", timeout: TimeSpan.FromSeconds(10));
+                response.EnsureSource(server, client);
 
                 var projectInfo = response.Payload;
-                Assert.Equal(projectName, projectInfo.Name);
-                Assert.Equal(2, projectInfo.Configurations.Count);
-                Assert.Contains("Debug", projectInfo.Configurations);
-                Assert.Contains("Release", projectInfo.Configurations);
+                Assert.Equal(projectName, projectInfo["Name"]);
+                Assert.Equal(2, projectInfo["Configurations"].Count());
+                Assert.Contains("Debug", projectInfo["Configurations"]);
+                Assert.Contains("Release", projectInfo["Configurations"]);
 
-                var frameworkShorNames = projectInfo.Frameworks.Select(f => f.ShortName);
-                Assert.Equal(3, frameworkShorNames.Count());
+                var frameworkShorNames = projectInfo["Frameworks"].Select(f => f["ShortName"].Value<string>());
+                Assert.Equal(2, frameworkShorNames.Count());
                 Assert.Contains("dnxcore50", frameworkShorNames);
                 Assert.Contains("dnx451", frameworkShorNames);
-                Assert.Contains("net46", frameworkShorNames);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ProtocolNegotiationTestData))]
+        public void DthStartup_ProtocolNegotiation(string flavor, string os, string architecture, int requestVersion, int expectVersion)
+        {
+            var projectName = "EmptyConsoleApp";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(projectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                client.SetProtocolVersion(requestVersion);
+
+                var response = client.DrainTillFirst(ProtocolManager.NegotiationMessageTypeName);
```

Comment:
That's because all the literal should be represent by a constant. It's the fault of production code.